Core/Foundation/Exception/Core_Foundation_Exception_Exception.php

```diff
@@ -23,11 +23,179 @@
+     */
+    public function __toString()
+    {
+        $data =  '<b>'.$this->message.'</b><br/>';
+        if ($afterMessageContent = $this->getAfterMessageContent()) {
+            $data .= '<i>Alternative used data given: "'.(string)$afterMessageContent.'"</i><br/>';
+        }
+        $data .= '<ul><li><b>Exception type:</b> '.get_class($this).'</li>';
+        if ($this->moduleToDeactivate != null) {
+            $data .= '<li><b>THIS MODULE HAS BEEN DEACTIVATED:</b> '.$this->moduleToDeactivate.'</li>';
+        }
+        $data .= '<li><b>Code #</b> '.$this->code.'</li>';
+        $data .= '<li><b>Occurred at line</b> '.$this->line.' <b>in file</b> '.$this->file.'</li>';
+        $data .= '<li><b>Time:</b> '.date('D, d M Y H:i:s').' / '.time().'</li>';
+        $data .= '</ul><br/>';
+
+        $data .= 'To send support information about this exception, please use this link:<br/>';
+        $data .= '<a href="#">TODO (This link will allow to send report to support team directly, in a future release)</a><br/><br/>';
+
+        if ($this->reportData != null && count($this->reportData) > 0) {
+            $data .= '<a name="'.$this->randomInstantiationKey.'_reporting_block"></a><a href="#'.$this->randomInstantiationKey.'_reporting_block"
+            onclick="$(this).next(\'.reporting_block\').toggle();">[+] <b>Reporting data</b></a>
+            <span class="reporting_block" style="display: none;">: <br/><ul>';
+            foreach ((array)$this->reportData as $reportKey => $reportItem) {
+                $data .= '<li><b>'.$reportKey.'</b>: '.(string)$reportItem.'</li>';
+            }
+            $data .= '</ul></span><br/>';
+        }
+
+        $data .= '<a name="'.$this->randomInstantiationKey.'_technical_block"></a><a href="#'.$this->randomInstantiationKey.'_technical_block"
+            onclick="$(this).next(\'.technical_block\').toggle();">[+] <b>Technical data</b></a>
+            <span class="technical_block" style="display: none;">: <br/><ul>';
+        try {
+            $request = Router::getLastRouterRequestInstance();
+            if ($request !== null) {
+                $data .= '<li><b>Route from module tracking:</b> '.$request->attributes->get('_route_from_module').'</li>';
+                $data .= '<li><b>Controller from module tracking:</b> '.$request->attributes->get('_controller_from_module').'</li>';
+            }
+        } catch (\Exception $e) {
+        }
+        $data .= '<li><b>File #</b> '.md5_file($this->file).'</li>';
+        $data .= '<li><b>URL:</b> <a href="http://'.$_SERVER['HTTP_HOST'].$_SERVER['REQUEST_URI'].'">http://'.$_SERVER['HTTP_HOST'].$_SERVER['REQUEST_URI'].'</a></li>';
+        $data .= '<li><b>Stack trace:</b> '.parent::getTraceAsString().'</li>';
+        if ($e = $this->getPrevious()) {
+            $data .= '<li><b>Previous Exception:</b> '.$e->__toString().'</li>';
+        }
+        $data .= '</ul></span><br/>';
```

Comment:
:-1: Not really, if the Template engine falls, then the Exceptions cannot be displayed, and then another Exception will be thrown by PHP... So in fine, you are not able to track the problem in the stackTrace...